Fiducial sensor for ros implemented

maxbader opened the pull request:
Hi

We enhanced the stage_ros pkg to publish fiducial sensor data.
In order to publish such data it was needed to add new messages the messages description is currently located within the stage_ros pkg. We are also in discussion with the common_msgs pkg maintainer to move the description into the sensor_msgs pkg (the brunch: without_stage_ros_messages implements this already).

I addtion we added a launch file for testing. Running:
`roslaunch stage_ros world.launch world:=fiducial`
should start a environment with fiducial.

We tested everything on Ubuntu 64bit 14.04 with ROS Jade, as well as on 16.04 with ROS Kinetic

We are looking forward to a merge 
![untitled](https://cloud.githubusercontent.com/assets/712440/15928672/9b4d89c0-2e48-11e6-9fc0-b9ecc806af84.jpg)

maxbader committed:
ros fiducial support added

maxbader committed:
Marker messages added

maxbader committed:
using common messages

maxbader committed:
message view direction modified

maxbader committed:
messages added

maxbader committed:
using new added messages

maxbader commented:
At the moment there is a discussion to add the marker message to the common_msgs pkg.
https://github.com/ros/common_msgs/issues/86

maxbader committed:
covariance from marker detection msg removed

maxbader committed:
covariance from marker detection msg removed

maxbader committed:
covariance from marker detection msg removed

wjwwood commented:
Ok, thanks for the pr. One of us will get to it as soon as we have time.

@gerkey FYI

wjwwood commented:
@maxbader also, when the discussion on `common_msgs` reaches a conclusion, please let us know on this thread. Thanks again.

maxbader committed:
moved to marker_msgs

Commit:
Adapted usage of marker_msgs::MarkerDetection (multiple ids with confidence)

Commit:
Enhanced fiducial world with more fiducials

maxbader committed:
marker type information added

maxbader committed:
using marker_msgs

Commit:
cosmetic: fixed copy&paste error

maxbader committed:
cleanup in world files

todorangrg committed:
changes:
- added kinematic constraints for differential drive (maximum velocity and acceleration of each wheel)

maxbader committed:
config update

maxbader committed:
config update

maxbader commented:
Hi 
We did now a lot of enhancements and code clean ups.
- The markers are working with a dependency to marker_msgs 
- We added the possibility to define motion contains on a differential drive

CodeFinder2 commented:
Regardless of the PR's content (which is great, BTW), I would advise against using `const_cast`'s, see (e.g.) https://github.com/ros-simulation/stage_ros/pull/36/commits/582784b654efa53242e3de4a538e54cf0a53b325#diff-85d0e0c7dc1f3bd834819d2137cba2c0R467

Instead, wouldn't is be possible to revisit the design of the stored member variables, possibly removing the `const` qualifier? Clearly, one would need to have a closer look at the impact. (I've done that in a completely modified/extended/refactored version of _stage_ros_ and it seems fine, at least in my version.)

wjwwood commented:
So is the conclusion to leave the marker messages out of `common_msgs` in favor of it's own package/repository?

That's fine, but we have to think carefully about adding this as a dependency of stage_ros, because stage_ros is part of what we call "desktop-full" and therefore is required to be released for any new ROS release's beta (like the upcoming ROS Lunar release) and so any dependencies of stage_ros also becomes part of that push. This just means that if we merge this and make marker_msgs a dependency of stage_ros then we are committing to ensuring it is released very early in the ROS distribution, and taking on responsibility for it if, in the future, you no longer have time to maintain that package. That's why it would have been nice to get it into common_msgs.

That being said, it's not a deal breaker, but it would make merging this an obvious decision.

maxbader commented:
Hi,

I started a discussion on ros/common_msgs#86 without success, because of this i created a package containing the marker message. There exists also a rviz plugin to visualize this message called marker_rviz_plugin. Both pkgs are released and already available via the ros pakage server.  I don't know maybe you have more influence on the ros comunity to move the message into the common msg pkg.
In order to ensure the maintains I can give you also the proper rights on the github repository but for the next years I am happy to keep the pkg maintained.

Greetings
Markus

PS.:  George (@todorangrg) will target the design issue stated by CodeFinder2 with a new commit within the next hours.

todorangrg committed:
removed const qualifier from StageRobot pointers (and also the const_casts)

todorangrg commented:
Hi,

I removed the const qualifier from the StageRobot pointers (and also all the const_casts). If you think I could change something else, please let me know. Also sorry for placing two enchantments in one pull request :)

Commit:
Fixed comments and indentation in world files.

wjwwood commented:
Sorry, I accidentally closed this when changing the branch layout of the repository. I've reopened it and updated the target base to be `lunar-devel` which is the new default branch. You can edit again to point at `indigo-devel`, which is what is used to release for ROS Indigo and Kinetic, if you like.

maxbader committed:
mixed map with fiducials added

maxbader committed:
cave world with fiducials update

maxbader committed:
Create README.md

maxbader committed:
Update README.md

maxbader committed:
ros2 stage_ros new start

maxbader committed:
Revert "ros2 stage_ros new start"

This reverts commit b7580d22eea5e94f344278c5609a0fe592ab11ef.

wjwwood commented:
Given the age of this pull request and the merge conflicts, I'm going to close this. Feel free to revive it or comment here on the closed issue if anyone disagrees.
